GDJS/Runtime/variablescontainer.ts

```diff
@@ -242,8 +242,9 @@ namespace gdjs {
         const variable = this._variables.get(variableName);
         const variableOwner = variable.getPlayerOwnership();
         if (
-          // Variable undefined.
-          variable.isUndefinedInContainer() ||
+          (!syncOptions.syncEverythingForWholeGameSaveState &&
```

Comment:
this doesn't seem correct. The && should encompass all the conditions of the existing if.
So something like `if (!syncOptions.forceSyncEverything && (variable.isUndefined() || xx || yy || ...))`

So that if forceSyncEverything is true then the `return` is skipped directly (in your case, just the condition `isUndefinedInContainer` is skipped